### DISCUSSION TITLE
Guidance on usage with React Native

### DISCUSSION THREAD

**Discussion by aeschylus** [nextauthjs/next-auth#6971] [Help]
### Question 💬

I'm aware of a number of issues that have been raised and closed over the last few years regarding usage with React Native. It seems a lot of work was done to improve compatibility, but the documentation still says that it is unsupported and not intended to be used with React Native.

I missed this detail when building my backend with Next.js, and now would like to find a way to not have to build a parallel auth stack (or replace everything). Can others share their experience with React Native and Next Auth? Is this possible with another library, or with a few overrides? What is the recommended way to use a custom client (particularly with Magic Link, Apple, and Google logins).

Thanks all!

### How to reproduce ☕️

https://github.com/nextauthjs/next-auth/issues/569#issuecomment-751393651

Here is an early attempt to override some things. Not sure how this has changed since then (it's been a few years now).

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by mattmcegg** [comment]
@aeschylus any progress on this? any suggestions?

**Comment by BayCo** [comment]
hey @aeschylus it's been much time i've been looking for answer to this question, i have also tracked all conversations by @stevesouth , how was your progress? is it still open question, or there is proper solution?

**Comment by vendramini** [comment]
Hey guys, do we have news here? How did you integrate RN?

**Reply by mattmcegg** [reply]
I ended up creating a second auth system just for mobile using passport.js that shares the same mongodb collections for Users and Accounts. Most annoying part was that I ended up creating all mobile specific API endpoints since there was a completely different token verification process. I guess this isn't really a solution but a workaround, but maybe will help someone stuck.

**Reply by vendramini** [reply]
Well, so I suppose it could be better to do all the workflow on passport.js and use it with Next.js instead of having 2 parallels workflow with the same goal?

**Reply by mattmcegg** [reply]
if you're just starting the project, yes, I think that would be the best path
